kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteFormModellerFormProvider.java

```diff
@@ -333,6 +337,81 @@ protected String filterXML(String document, String lang, String deploymentId, Ma
         return document;
     }
 
+    protected String attachSubForms(String document, String deploymentId) {
```

Comment:
I don't think these will make it easier to read. Reading xml and going through some nodes is similar but there is much more work done in filterXml than in attachSubforms and that was the main reason to not merge/reuse it.